판매자 회원가입 비밀번호 검증 라벨 추가 및 리팩토링

1000hyehyang opened the pull request:
## ⭐️ Issue Number
- #19

## 🚩 Summary
![image](https://github.com/user-attachments/assets/4e69857c-6bc6-4ae8-af5d-d5977132ae15)

- 회원가입 폼을 재사용 가능한 FormField 컴포넌트로 리팩토링하여 코드 중복 제거 및 일관성 향상
- 비밀번호 조건 표시를 PasswordStrengthIndicator 컴포넌트로 분리하여 관심사 분리 및 재사용성 개선
- useSignupForm 훅으로 폼 상태 관리 및 유효성 검사 로직을 분리하여 성능 최적화
- 헬퍼 텍스트와 글자 수를 한 줄에 양 끝 배치하는 레이아웃 개선으로 UX 향상
- TypeScript 타입 정의를 별도 파일로 분리하여 타입 안정성 강화

## 🛠️ Technical Concerns

- FormField 컴포넌트의 props 인터페이스 설계 시 재사용성과 확장성 고려
- useSignupForm 훅에서 상태 업데이트 최적화를 위한 useCallback 사용
- 비밀번호 조건 검증 로직의 성능 최적화
- 타입 정의 시 유니온 타입과 제네릭을 활용한 타입 안전성 확보

## 🙂 To Reviewer
@taehyun32 
백엔드 유효성 검증에 맞춰서 프론트도 수정했습니다! 하는김에 리팩토링으로 구조 분리를 해봤어요!

## 📋 To Do

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * 새로운 폼 필드 UI 컴포넌트와 비밀번호 강도 표시 기능이 추가되었습니다.
  * 회원가입 폼 입력값 자동 포맷팅(전화번호, 사업자번호)이 적용되었습니다.

* **Refactor**
  * 회원가입 폼의 상태 관리와 입력 핸들러가 커스텀 훅으로 분리되어 코드 구조가 개선되었습니다.
  * 폼 스타일이 일관성 있게 통일되었습니다.

* **Style**
  * 폼 요소(입력창, 버튼, 체크박스 등)의 스타일이 개선되어 UI가 더욱 일관되고 보기 좋아졌습니다.

* **Chores**
  * 웹킷 브라우저 자동완성 스타일이 개선되어 입력 필드의 기본 노란색 배경이 제거되었습니다.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

1000hyehyang committed:
refactor: 입력폼 컴포넌트 분리 및 비밀번호 조건 표시 추가

- FormField 컴포넌트로 재사용 가능한 폼 필드 구현
- PasswordStrengthIndicator 컴포넌트로 비밀번호 조건 표시 분리
- useSignupForm 훅으로 상태 관리 및 유효성 검사 로직 분리
- 헬퍼 텍스트와 글자 수를 한 줄에 양 끝 배치하는 레이아웃 개선
- 폼 필드 타입, 약관 동의 타입, 헬퍼 텍스트 타입 등 타입 정의 추가
- 비밀번호 확인 및 브랜드명 필드의 헬퍼 텍스트 일관성 개선
- 코드 구조 개선으로 유지보수성 및 재사용성 향상

taehyun32 reviewed (approved):
비밀번호, 브랜드명, 약관 동의 등 각 항목별로 명확하게 조건을 나눠서 검증 로직이 이해하기 쉬운 것 같습니다.
검증 함수를 따로 분리해서 로직 흐름도 깔끔하고 테스트나 재사용에도 좋아 보입니다.
입력 관리부터 유효성 검사, 제출까지 훅 하나로 정리돼 있어서 폼 관련 로직이 잘 모여 있어 보입니다. 수고하셨습니다 🙌

1000hyehyang committed:
refactor: 전화번호 입력 시 숫자만 저장하도록 수정

사업자 등록번호와 일관성을 맞춰 숫자만 저장하도록 변경

1000hyehyang commented on src/app/seller-signup/page.tsx:
```diff
@@ -161,251 +59,229 @@ export default function SellerSignUpPage() {
                 type="text"
                 placeholder="123-45-67890"
                 value={formatBusinessNumber(signupFormData.businessNumber)}
                 onChange={(e) =>
-                  handleInputChange('businessNumber', e.target.value.replace(/[^0-9]/g, ''))
+                  handleInputChange(
+                    'businessNumber' as FormFieldType,
+                    e.target.value.replace(/[^0-9]/g, ''),
+                  )
```
반영했다

1000hyehyang committed:
design: 브라우저 자동완성 배경 제거

- input:-webkit-autofill 스타일 추가
- 자동완성 시 배경색을 디자인 시스템 색상으로 강제 적용
- 입력 필드 시각적 일관성 개선
